gobblin-core/src/main/java/org/apache/gobblin/source/extractor/partition/Partitioner.java

```diff
@@ -407,9 +429,30 @@ protected long getHighWatermark(ExtractType extractType, WatermarkType watermark
         highWatermark = this.getAppendHighWatermark(extractType);
       }
     }
+
+    // Use fixed high watermark in bootstrapping mode
+    if (isBootStrapWithFixedHighWatermark()) {
+      hasUserSpecifiedHighWatermark = true;
+      if (this.state.contains(ConfigurationKeys.SOURCE_QUERYBASED_END_VALUE)) {
+        highWatermark = this.state.getPropAsLong(ConfigurationKeys.SOURCE_QUERYBASED_END_VALUE);
+        LOG.info("Bootstrapping mode: persisted high watermark " + highWatermark);
+      } else {
+        this.state.setProp(ConfigurationKeys.SOURCE_QUERYBASED_END_VALUE, highWatermark);
```

Comment:
Is this state getting persisted and propagated to the next run? If it does then what happens when user changes the configuration? Will the config change override the state?